### PR TITLE
feat(xattr_error): handle xattr set fail

### PR DESCRIPTION
**Pull request by abhilashshetty04**

### CODE REVIEW

File: io-engine/src/grpc/v1/replica.rs [hunk 1]
```diff
@@ -253,7 +254,14 @@ impl ReplicaRpc for ReplicaService {
...
                         debug!("created lvol {:?}", lvol);
                         if let Some(v) = &args.entity_id {
-                            let _ = lvol.set_blob_attr("entity_id", v.to_string(), true).await;
+                            if let Err(e) = lvol.set_entity_id(v.to_string()).await {
+                                return Err(
+                                    LvsError::SetXAttr {
+                                        source: e.to_errno(),
+                                        attr: "entity_id".to_string(),
+                                        name: args.uuid
+                                    })
```
Comment (tiagolobocastro) [inline io-engine/src/grpc/v1/replica.rs]: can't set_entity_id return LvsError already?

File: io-engine/src/lvs/lvs_lvol.rs [hunk 2]
```diff
@@ -572,6 +572,9 @@ pub trait LvsLvol: LogicalVolume + Share {
         prop: PropValue,
     ) -> Result<(), Error>;
 
+    /// Sets entity_id with given arg to Lvol xattr.
+    async fn set_entity_id(&self, id: String) -> Result<(), Error>;
```
Comment (tiagolobocastro) [inline io-engine/src/lvs/lvs_lvol.rs]: Ah looks like we have existing property below, I think you should add entityId as another enum variant to PropValue

File: io-engine/src/grpc/v1/replica.rs (hunk 1 above)
Comment (tiagolobocastro) [reply io-engine/src/grpc/v1/replica.rs]: Also btw shouldn't this be done as part of the lvol creation itself?

Comment (abhilashshetty04) [reply io-engine/src/grpc/v1/replica.rs]: Sure will do.

File: io-engine/src/lvs/lvs_lvol.rs [hunk 3]
```diff
@@ -842,6 +846,33 @@ impl LvsLvol for Lvol {
                     }),
                 }
             }
+
+            PropName::EntityId => {
+                let name = "entity_id".to_string().into_cstring();
+                let mut value: *const libc::c_char =
```
Comment (tiagolobocastro) [inline io-engine/src/lvs/lvs_lvol.rs:852]: note for later, seems a lot of duplicated code around this area (existing), maybe can improve later

File: io-engine/src/lvs/lvs_lvol.rs [hunk 4]
```diff
@@ -943,6 +974,23 @@ impl LvsLvol for Lvol {
                     name: self.name(),
                 })?;
             }
+            PropValue::EntityId(id) => {
+                let name = "entity_id".to_string().into_cstring();
```
Comment (tiagolobocastro) [inline io-engine/src/lvs/lvs_lvol.rs]:
```suggestion
                let name = PropName::from(&prop).to_string().into_cstring();
```

**Review by tiagolobocastro — APPROVED** [review]

File: io-engine/src/lvs/lvs_lvol.rs (hunk 3 above)
Comment (abhilashshetty04) [reply io-engine/src/lvs/lvs_lvol.rs:852]: Yes i noticed. The bit where we call spdk method for xattr ops right?

**Review by dsharma-dc — APPROVED** [review]

**Comment by abhilashshetty04** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors cancel

File: io-engine/src/lvs/lvs_lvol.rs (hunk 4 above)
Comment (tiagolobocastro) [reply io-engine/src/lvs/lvs_lvol.rs]: This is marked as resolved but not addressed..
Please do not mark comments as resolved without making the changes or discuss the change.

File: io-engine/src/lvs/lvs_lvol.rs (hunk 3 above)
Comment (tiagolobocastro) [reply io-engine/src/lvs/lvs_lvol.rs:852]: Yes indeed, we can have a look at it at a later time

File: io-engine/src/lvs/lvs_lvol.rs (hunk 4 above)
Comment (abhilashshetty04) [reply io-engine/src/lvs/lvs_lvol.rs]: Sorry missed it. Done now

**Comment by abhilashshetty04** [comment]
bors merge

**Pull request merged**